rustls/src/error.rs

```diff
@@ -636,7 +634,11 @@ mod other_error {
     ///
     /// Enums holding this type will never compare equal to each other.
     #[derive(Debug, Clone)]
-    pub struct OtherError(#[cfg(feature = "std")] pub Arc<dyn StdError + Send + Sync>);
+    pub struct OtherError(
+        // XXX XXX REMOVING THIS FEATURE CONDITION LEADS TO BUILD ERRORS IN rustls/src/webpki/mod.rs - XXX TBD HOW TO RESOLVE THESE ???
+        #[cfg(feature = "std")]
```

Comment:
This is what stopped me from finishing & merging these updates earlier this week.

In general, I would peronally consider this kind of API incompatibility between std & no-std to be a bug. This makes it harder to write feature-independent code.